mkosi/backend.py

```diff
@@ -244,24 +244,24 @@ class MkosiConfig:
     """
 
     verb: Verb
-    cmdline: list[str]
+    cmdline: Sequence[str]
```

Comment:
This change is not a good idea, because it gives us less type safety instead of more.  The one thing is, that we actually do fill this with lists, which have a different variance than `Sequence`  and nothing can guarantee that they are not changed (i.e. even if the dataclass were frozen, it would be the list changing instead of the `MkosiConfig` instance). The other problem is that `str` is an instnace of `Sequence[str]`, so with this change bare strings would properly typecheck wherever we don't expect it.

Reply:
Dropped